components/app-proxy/vm.tf

```
@@ -6,78 +6,65 @@ module "tags" {
   builtFrom   = var.builtFrom
 }
 
-resource "azurerm_resource_group" "this" {
+resource "azurerm_resource_group" "app_proxy_rg" {
   name     = "${var.product}-${var.env}-rg"
   location = var.location
 
   tags = module.tags.common_tags
 }
 
-# TODO store in a vault
-resource "random_password" "this" {
-  length = 30
-}
-
-resource "azurerm_subnet" "this" {
-  name = "app-proxy"
-
+resource "azurerm_subnet" "app_proxy_subnet" {
```

https://www.terraform-best-practices.com/naming#resource-and-data-source-arguments

either
```suggestion
resource "azurerm_subnet" "app_proxy" {
```

or
```suggestion
resource "azurerm_subnet" "this" {
```